src/App.js

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react';
+import {
+  BrowserRouter as Router, Routes, Route, Link,
+} from 'react-router-dom';
+import BookList from './components/bookList';
+import BookForm from './components/bookform';
+import CategoriesPage from './components/categories/CategoriesPage';
+import './App.css';
+
+function App() {
+  const [books, setBooks] = useState([]);
+
+  const handleAddBook = (newBook) => {
+    const bookWithId = { ...newBook, id: Date.now() };
+    setBooks([...books, bookWithId]);
+  };
+
+  const handleDeleteBook = (bookId) => {
+    const updatedBooks = books.filter((book) => book.id !== bookId);
+    setBooks(updatedBooks);
+  };
+
+  return (
+    <Router>
+      <div className="app">
+        <nav>
+          <ul>
+            <li>
+              <Link to="/">Home</Link>
+            </li>
+            <li>
+              <Link to="/categories">Categories</Link>
+            </li>
+          </ul>
+        </nav>
```

Comment:
You did well with your project, please I'd suggest that for easy code maintenance and best practice, you should use more of the components in your React and especially in the ` App.js ` file. I'd suggest you create a separate component for the Navbar and then import the component in your `App.js` file, to enable clean code and best javascript practices while ensuring you resist redundancy in your code base 😄 

Something like this 👍 

![image](https://github.com/kerlos-bahaa/Bookstore-react/assets/20537712/6d222b2d-6573-42ad-8dd1-7d4e1424513e)

Reply:
@usorfaitheloho  i don't think that in requirement is it optional

Reply:
Hello @kerlos-bahaa 😄 ,  you maintaining a clean code base and ensuring you avoid future repetition of code is a best coding practice that should be followed. It is best to use components so that they can be imported when you need them